java/ql/src/utils/flowtestcasegenerator/FlowTestCaseUtils.qll

```diff
@@ -69,11 +64,12 @@ string getZero(PrimitiveType t) {
  * Holds if `c` may require disambiguation from an overload with the same argument count.
  */
 predicate mayBeAmbiguous(Callable c) {
-  exists(Callable other, string package, string type, string name |
-    c.hasQualifiedName(package, type, name) and
+  exists(Callable other, Callable override, string package, string type, string name |
+    override = [c, c.(Method).getASourceOverriddenMethod*()] and
```

Comment:
Since you're using the reflexive transitive closure here, I think you don't need to add `c`, it should be included by `c.(Method).getASourceOverriddenMethod*()` already:

```suggestion
    override = c.(Method).getASourceOverriddenMethod*() and
```

Reply:
`c` can also be a `Constructor`, which diesn't have `getASourceOverridenMethod`, so I think `c` is still necassary for that case.

Reply:
Oh, good point! 👍

Reply:
(Maybe using the non-reflexive transitive closure would be a bit clearer then, but anyway)